library/src/G4PhononScattering.cc

```diff
@@ -89,20 +90,11 @@ G4VParticleChange* G4PhononScattering::PostStepDoIt( const G4Track& aTrack,
     }
   }
 
-  // Assign new wave vector direction to track (ought to happen later!)
-  auto trkInfo = G4CMP::GetTrackInfo<G4CMPPhononTrackInfo>(aTrack);
-  trkInfo->SetWaveVector(newK);
-
-  // Set velocity and direction according to new wave vector direction
-  G4double vgrp = theLattice->MapKtoV(mode, newK);
-  G4ThreeVector vdir = theLattice->MapKtoVDir(mode, newK);
-  RotateToGlobalDirection(vdir);
+  // Update particleChange with new wavevector and Vg
+  FillParticleChange(aParticleChange, aTrack, newK);
 
   if (verboseLevel>1)
-    G4cout << " new vgrp " << vgrp << " along " << vdir << G4endl;
-
-  aParticleChange.ProposeMomentumDirection(vdir);
-  aParticleChange.ProposeVelocity(vgrp);
+    G4cout << " new vgrp " << aParticleChange.GetVelocity() << " along " << *aParticleChange.GetMomentumDirection() << G4endl;
```

Comment:
`aParticleChange` is an object, not a pointer.  Should not have "*", and it shouldn't have compiled this way.